source/slang/slang-ir-inst-defs.h

```diff
@@ -235,6 +236,8 @@ INST(makeVector, makeVector, 0, 0)
 INST(MakeMatrix, makeMatrix, 0, 0)
 INST(makeArray, makeArray, 0, 0)
 INST(makeStruct, makeStruct, 0, 0)
+INST(MakeTuple, makeTuple, 0, 0)
+INST(GetTupleElement, getTupleElement, 2, 0)
```

Comment:
Can `getTupleElement` just use the existing `getElement` (or is it `extractElement`?) instruction? I guess the reason to have a distinct case is that this operation is index-based (like array `getElement` and unlike struct ` getField`), but the type of the result depends on the index/key (like struct `getField` but unlike array `getElement`).

Reply:
This makes it easier for the lowering pass to identify the instruction. Otherwise it needs to also examine the type of the first operand of the `getElement` inst to tell if it is dealing with a tuple or not.

Reply:
Sounds good.